components/style/values/specified/mod.rs

```diff
@@ -310,20 +310,48 @@ impl ToCss for BorderRadiusSize {
     }
 }
 
-#[derive(Clone, PartialEq, PartialOrd, Copy, Debug)]
+#[derive(Clone, PartialEq, Copy, Debug)]
 #[cfg_attr(feature = "servo", derive(HeapSizeOf, Deserialize, Serialize))]
-/// An angle, normalized to radians.
+/// An angle consisting of a value and a unit.
 pub struct Angle {
-    radians: CSSFloat,
+    value: CSSFloat,
+    unit: AngleUnit,
     was_calc: bool,
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf, Deserialize, Serialize))]
+/// A unit used together with an angle.
+pub enum AngleUnit {
+    /// Degrees, short name "deg".
+    Degree,
+    /// Gradians, short name "grad".
+    Gradian,
+    /// Radians, short name "rad".
+    Radian,
+    /// Turns, short name "turn".
+    Turn,
+}
+
+impl ToCss for AngleUnit {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        use self::AngleUnit::*;
+        dest.write_str(match *self {
+            Degree => "deg",
+            Gradian => "grad",
+            Radian => "rad",
+            Turn => "turn",
+        })
+    }
+}
+
 impl ToCss for Angle {
     fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
         if self.was_calc {
             dest.write_str("calc(")?;
         }
-        write!(dest, "{}rad", self.radians)?;
+        write!(dest, "{}", self.value)?;
```

Comment:
nit: you could also use `self.value.to_css(dest)` if you preferred, it'd desugar to the same though, so your call.